src/kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs

```diff
@@ -1304,17 +1394,21 @@ fn normalize_type(ty: Ty) -> Ty {
 
 impl<'tcx> GotocCtx<'tcx> {
     /// A pointer to the mir type should be a thin pointer.
+    /// Use thin pointer if the type is sized or if the resulting pointer has no metadata.
     pub fn use_thin_pointer(&self, mir_type: Ty<'tcx>) -> bool {
         // ptr_metadata_ty is not defined on all types, the projection of an associated type
         let (metadata, _check_is_sized) = mir_type.ptr_metadata_ty(self.tcx, normalize_type);
         return !self.is_unsized(mir_type) || metadata == self.tcx.types.unit;
     }
     /// A pointer to the mir type should be a slice fat pointer.
+    /// We use a slice fat pointer if the metadata is the slice length (type usize).
     pub fn use_slice_fat_pointer(&self, mir_type: Ty<'tcx>) -> bool {
         let (metadata, _check_is_sized) = mir_type.ptr_metadata_ty(self.tcx, normalize_type);
         return metadata == self.tcx.types.usize;
     }
     /// A pointer to the mir type should be a vtable fat pointer.
+    /// We use a vtable fat pointer if this is a fat pointer to anything that is not a slice ptr.
+    /// I.e.: The metadata is not length (type usize).
```

Comment:
Is it possible to check that it is a vtable, rather than it not being the other two options?

Reply:
I can try to add an assert. I just added a comment while figuring out what it was doing. For this PR, do you prefer if I remove the comments?

Reply:
I think the comments are useful